File: Runtime/CoreLib/Globalization/CultureInfo.cs
// CultureInfo.cs
// Script#/Libraries/CoreLib
// This source code is subject to terms and conditions of the Apache License, Version 2.0.
//

using System.Runtime.CompilerServices;

namespace System.Globalization {
	[Imported(ObeysTypeSystem = true)]
	[ScriptNamespace("ss")]
	public sealed class CultureInfo : IFormatProvider {
		private CultureInfo() {
		}

		[PreserveCase]
		[IntrinsicProperty]
		public static CultureInfo CurrentCulture {
			get {
				return null;
			}
		}

		[IntrinsicProperty]
		public DateTimeFormatInfo DateTimeFormat {
			get {
				return null;
			}
		}

		[PreserveCase]
		[IntrinsicProperty]
		public static CultureInfo InvariantCulture {
			get {
				return null;
			}
		}

		[IntrinsicProperty]
		public string Name {
			get {
				return null;
			}
		}

		[IntrinsicProperty]
		public NumberFormatInfo NumberFormat {
			get {
				return null;
			}
		}

		public object GetFormat(Type formatType) {
			return null;
		}
	}
}

File: Runtime/CoreLib.Script/CultureInfo.js
///////////////////////////////////////////////////////////////////////////////
// CultureInfo

var ss_CultureInfo = ss.CultureInfo = ss.mkType(ss, 'ss.CultureInfo',
	function#? DEBUG CultureInfo$##(name, numberFormat, dateTimeFormat) {
		this.name = name;
		this.numberFormat = numberFormat;
		this.dateTimeFormat = dateTimeFormat;
	},
	{
		getFormat:  function#? DEBUG CultureInfo$getFormat##(type) {
			switch (type) {
				case ss_NumberFormatInfo: return this.numberFormat;
				case ss_DateTimeFormatInfo: return this.dateTimeFormat;
				default: return null;
			}
		}
	}
);

ss.initClass(ss_CultureInfo, null, [ss_IFormatProvider]);

ss_CultureInfo.invariantCulture = new ss_CultureInfo('en-US', ss_NumberFormatInfo.invariantInfo, ss_DateTimeFormatInfo.invariantInfo);
ss_CultureInfo.currentCulture = ss_CultureInfo.invariantCulture;
